tasks/tasks/causemos_processors.py

```diff
@@ -140,35 +223,156 @@ def generate_index_model_weights(
     for key, value in weighted_sums.items():
         result_percent[key] = value / total_sums
 
-    print(result_percent)
+    return result_percent.values()
+
+
+# FOr v2 maybe later:
+# Option A Impl: TODO
+    # def calc_highest_ratios_4pc_ea(explained_variance_ratios: List):
+    #     pass
+
+    # # Option B Impl: TODO
+    # def calc_highest_ratios_90pc_total(explained_variance_ratios: List):
+    #     pass
+
+
+def pca_to_weights_v2(pca_details: Tuple):
+    """
+    Joel's Average pca-weights-algorithm
+    as described in doc: `PCA: Feature Contributions Using Control Dataset`
+
+    This is using a Option A above (filters out any PC with explained variance less than 4%)
+    which has disadvantages under even-PC-explained-variance scenarios
+
+    Option B would use the same approach as what v3 uses, but has not been implemented here.
+    """
+    (pca_components_list, pca_components_count, pca_explained_ratios) = pca_details
+
+    highest_variance_ratios = list(filter(lambda x: x > 0.04, map(roundToThree, pca_explained_ratios)))
+    weight_matrix = [[0] * pca_components_count for i in range(0, len(highest_variance_ratios))]
+    amount_relevant = len(highest_variance_ratios)
+
+    for pc_index, principle_component in enumerate(pca_components_list):
+        for feature_index, value in enumerate(principle_component):
+            relative_value = value * pca_explained_ratios[pc_index]
+
+            if pc_index < amount_relevant:
+                weights_row = weight_matrix[pc_index]
+                weights_row[feature_index] = weights_row[feature_index] + relative_value
+
+    matrix_percents = list(map(calc_percents, weight_matrix))
+    matrix_averages = [(sum(v)/amount_relevant) for i, v in enumerate(zip(*matrix_percents))]
+
+    return matrix_averages
+
+
+def pca_to_weights_v3(pca_details: Tuple, target_pca_explained_sum=0.7):
```

Comment:
@brandomr this is the algorithm you shared on slack and we went through recently. It is the one wired to the whole flow as of the current state of this PR.

I added the other v1, v2 version functions in order for us to test/toggle and compare as well.